aws-lc-rs/src/lib.rs

```diff
@@ -124,10 +124,49 @@ mod ec;
 mod ed25519;
 mod endian;
 mod evp_pkey;
+mod fips;
 pub mod iv;
 mod ptr;
 mod rsa;
 
+#[cfg(feature = "fips")]
+pub use crate::fips::reset_fips_service_status;
+
+pub use crate::fips::{get_fips_service_status, FipsServiceStatus};
+
+/// Use `check_fips_service_status` to wrap an **aws-lc-rs** cryptographic function, and verify
+/// that the operation was approved by the FIPS module. By default this macro will reset the FIPS module
+/// status if the operation was not approved. You must use `FipsServiceStatus` to determine if
+/// the operation was not approved. If the crate was not built with the `fips` feature, this function will
+/// always wrap the function result with `FipsServiceStatus::NotEnabled`.
+#[macro_export]
+macro_rules! check_fips_service_status {
```

Comment:
Rather than a macro it might be a bit more ergonomic to do a FIPS "guard" which triggers a check when it goes out of scope:

```
{
    let _fips_guard = aws_lc_rs::fips::guard();

    // do crypto here

    // fips::Guard Drop impl is called, which will check the FIPS status
}
```